tabular/src/autogluon/tabular/models/knn/_knn_loo_variants.py

```diff
@@ -1,6 +1,16 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
 import numpy as np
 from scipy import stats
-from sklearn.neighbors import KNeighborsClassifier as _KNeighborsClassifier, KNeighborsRegressor as _KNeighborsRegressor
+try:
+    # TODO: Difficult to add user switch for daal usage without duplicating code (Sklearn and daal versions of KNN for predict_loo
+    from daal4py.sklearn.neighbors import KNeighborsClassifier as _KNeighborsClassifier, KNeighborsRegressor as _KNeighborsRegressor
+    # daal backend for KNN seems to be 20-40x+ faster than native sklearn with no downsides.
+    logger.log(15, '\tUsing daal4py KNN backend...')
+except:
+    from sklearn.neighbors import KNeighborsClassifier as _KNeighborsClassifier, KNeighborsRegressor as _KNeighborsRegressor
```

Comment:
maybe print tip to install daal4py in order to get faster KNN models?

Reply:
will do this in future PR once we have a better idea of how prominently we want to encourage normal users to install daal. It's possible we add it to `autogluon.tabular[all]` and the default `autogluon` install, although the size of the package and its dependencies are somewhat large (~500MB - 1 GB).